qiime2/sdk/usage.py

```diff
@@ -8,60 +8,194 @@
 
 import abc
 import re
-import types
+from types import MappingProxyType
+from typing import Tuple, Callable, Optional, Union, List, Set, Dict
 
 from qiime2 import sdk, metadata
+from qiime2.core.type import MethodSignature, PipelineSignature
 
-# TODO: docstrings
+# Is Metadata sufficient for typing, or should we specify MetadataColumn
```

Comment:
Metadata is sufficient, since we don't allow MetadataColumn as a primitive in usage API.